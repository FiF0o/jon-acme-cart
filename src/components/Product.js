import React from 'react'
import PropTypes from 'prop-types';

export const Product = ({ productCode, name, price, productImg }) => (
  <div className="mdc-card demo-basic-with-header">
    <div className="demo-card__primary">
      <h2 className="demo-card__title mdc-typography mdc-typography--headline6">{ name }</h2>
      <h3 className="demo-card__subtitle mdc-typography mdc-typography--subtitle2">{ productCode }</h3>
    </div>
    <div className="mdc-card__primary-action demo-card__primary-action" tabIndex="0">
      <div 
        className="mdc-card__media mdc-card__media--16-9 demo-card__media"
        style={{
          backgroundImage: `url(${productImg})`
        }}
        ></div>
      <div className="demo-card__secondary mdc-typography mdc-typography--body2">Â£{ price }</div>
    </div>
  </div>
)

Product.propTypes = {
  productCode: PropTypes.string.isRequired,
  productImg: PropTypes.string,
  name: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired
}