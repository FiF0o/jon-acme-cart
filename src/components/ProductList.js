import React from 'react';
import PropTypes from 'prop-types';

import { Product } from './Product'

// for simplicity of the exercise we say ee receive the basket directly
export const ProductList = ({ basket }) =>
  <div>
    {
      basket.map(item =>
        <Product
          key={item.productCode}
          productCode={item.productCode}
          productImg={item.productImg}
          name={item.name}
          price={item.price}
        />
      )
    }
  </div>

Product.propTypes = {
  basket: PropTypes.arrayOf(PropTypes.instanceOf(Product)),
}