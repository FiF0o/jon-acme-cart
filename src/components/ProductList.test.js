
import React from 'react';
import { shallow } from 'enzyme';

import { ProductList } from './ProductList'
import { Product } from './Product'

describe('ProductList component', () => {
  it('should render the list of products correctly', () => {
    const basket = [
      {
        productCode: 'ABCD',
        name: 'ABC',
        price: 4,
        productImg: 'some/dummy/imgLink'
      },
      {
        productCode: 'ABC',
        name: 'ABC',
        price: 4,
        productImg: 'some/dummy/imgLink'
      }
    ]

    const wrapper = shallow(<ProductList basket={basket} />)
    expect(wrapper.find(Product).length).toBe(2)
    expect(wrapper).toMatchSnapshot()
  })
})