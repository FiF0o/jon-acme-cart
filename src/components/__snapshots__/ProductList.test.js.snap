// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList component should render the list of products correctly 1`] = `
<div>
  <Product
    key="ABCD-0"
    name="ABC"
    price={4}
    productCode="ABCD"
    productImg="some/dummy/imgLink"
  />
  <Product
    key="ABC-1"
    name="ABC"
    price={4}
    productCode="ABC"
    productImg="some/dummy/imgLink"
  />
</div>
`;
